clang/lib/AST/ExprConstant.cpp

```diff
@@ -10895,6 +10899,132 @@ bool VectorExprEvaluator::VisitUnaryOperator(const UnaryOperator *E) {
   return Success(APValue(ResultElements.data(), ResultElements.size()), E);
 }
 
+static bool EvaluateVectorOrLValue(APValue &Result, EvalInfo &Info,
```

Comment:
This interface to this function seems really odd for the name?  Is the name inaccurate, or is this using 'false' to mean 'failure' AND 'not a vector'?  

It seems to me that this needs to be something like "EvaluateLValueVector" or something?

Reply:
> is the name inaccurate, or is this using 'false' to mean 'failure' AND 'not a vector'?
Indeed, false is being treated as a failure and not a vector.
> It seems to me that this needs to be something like "EvaluateLValueVector" or something?
I'm not entirely sure if the way I implemented the handling of an lvalue as a parameter is even the correct way to go with, so it might very well be that the reason it sticks out is because I'm doing something wrong. I guess I'll have to wait for someone who has dealt with lvalues in constexpr to tell me where I went wrong

Reply:
I think that these errors should be covered by convertvector.c, I'll add one of the missing diagnostics you pointed out. One already exists

Reply:
Went with the EvaluateAsRValue